var/spack/repos/builtin/packages/coreneuron/package.py

```diff
@@ -19,7 +19,9 @@ class Coreneuron(CMakePackage):
     git      = "https://github.com/BlueBrain/CoreNeuron"
 
     version('develop', branch='master')
-    version('1.0b', tag="1.0b", submodules=True, preferred=True)
+    # 1.0.0 > 1.0b > 1.0 as far as Spack is concerned
```

Comment:
So `@1.0` should resolve to `@1.0.0`, I guess…

Reply:
I think with the current version then there is no v1.0, trying to install it is an error:
```
$ spack install coreneuron@1.0
...
==> Error: Unable to parse extension from https://github.com/BlueBrain/CoreNeuron.

If this URL is for a tarball but does not include the file extension
in the name, you can explicitly declare it with the following syntax:

    version('1.2.3', 'hash', extension='tar.gz')

If this URL is for a download like a .jar or .whl that does not need
to be expanded, or an uncompressed installation script, you can tell
Spack not to expand it with the following syntax:

    version('1.2.3', 'hash', expand=False)
```
which is annoying, because the release is branded as 1.0 not 1.0.0. Is there a way to say "treat @1.0 as @1.0.0"?

Reply:
It used to be that way, but would lead to inconsistencies. Basically, `%intel` would identify as `19.0.254.123`, but the spec would be `19.0.254`… and we would have a mix of paths, depending how much of the version one specified.

Reply:
Fair enough, thanks for the explanation.